configure.ac

```diff
@@ -652,18 +706,25 @@ if test "x$enable_md2" = "xyes"; then
 	enable_md2=yes
 else
 	enable_md2=no
-	CFLAGS="$CFLAGS -DNOMD2"
+	AC_DEFINE([NOMD2])
 fi
 AM_CONDITIONAL([ENABLE_MD2], [test "x$enable_md2" = "xyes"])
 
 AC_SUBST([pkcsslotd_user])
 AC_SUBST([pkcs_group])
+AC_DEFINE([PKCS64])
+AC_DEFINE([_XOPEN_SOURCE], [600], [POSIX 2004])
+if test "x$build_aix" = "xyes"; then
+	CFLAGS="$CFLAGS -qinfo=pro:cns:lan:nocmp:nocnv:nogot:notrd -qalign=natural -qarch=pwr7 -qsuppress=1506-1527 -qlanglvl=stdc99 -qsrcmsg -lbsd -q64"
+else
+	CFLAGS="$CFLAGS -Wall -Wextra"
+	CFLAGS+=" -std=c99 -Werror=implicit-int -Werror=implicit-function-declaration -Werror=int-conversion -Werror=strict-prototypes -Werror=old-style-definition -pedantic"
+fi
 
-CFLAGS="$CFLAGS -DPKCS64 -D_XOPEN_SOURCE=600 -Wall -Wextra"
+AC_DEFINE_UNQUOTED([PKCSSLOTD_USER], "$pkcsslotd_user")
+AC_DEFINE_UNQUOTED([PKCS_GROUP], "$pkcs_group")
 
-CFLAGS+=' -DCONFIG_PATH=\"$(localstatedir)/lib/opencryptoki\" -DSBIN_PATH=\"$(sbindir)\" -DLIB_PATH=\"$(libdir)\" -DLOCKDIR_PATH=\"$(lockdir)\" -DOCK_CONFDIR=\"$(sysconfdir)/opencryptoki\" -DOCK_LOGDIR=\"$(logdir)\"'
-CFLAGS+=' -DPKCSSLOTD_USER=\"$(pkcsslotd_user)\" -DPKCS_GROUP=\"$(pkcs_group)\"'
-CFLAGS+=" -std=c99 -Werror=implicit-int -Werror=implicit-function-declaration -Werror=int-conversion -Werror=strict-prototypes -Werror=old-style-definition -pedantic"
+CFLAGS="$CFLAGS"' -DCONFIG_PATH=\"$(localstatedir)/lib/opencryptoki\" -DSBIN_PATH=\"$(sbindir)\" -DLIB_PATH=\"$(libdir)\" -DRUN_PATH=\"$(runstatedir)\" -DLOCKDIR_PATH=\"$(lockdir)\" -DOCK_CONFDIR=\"$(sysconfdir)/opencryptoki\" -DOCK_LOGDIR=\"$(logdir)\"'
```

Comment:
I guess those could also be done using AC_DEFINE_UNQUOTED?

Reply:
Nope, I tried that already. We need these variables to be expanded by automake, not autoconf - they need to survive in their variable form until automake can substitute them when `make` itself is called.

Reply:
Ah OK, I see.